sdk/search/Azure.Search.Documents/src/Generated/KnowledgeSourcesRestClient.cs

```diff
@@ -409,5 +409,75 @@ public Response<KnowledgeSource> Create(KnowledgeSource knowledgeSource, Cancell
                     throw new RequestFailedException(message.Response);
             }
         }
+
+        internal HttpMessage CreateGetStatusRequest(string sourceName)
+        {
+            var message = _pipeline.CreateMessage();
+            var request = message.Request;
+            request.Method = RequestMethod.Get;
+            var uri = new RawRequestUriBuilder();
+            uri.AppendRaw(_endpoint, false);
+            uri.AppendPath("/knowledgesources('", false);
+            uri.AppendPath(sourceName, true);
+            uri.AppendPath("')/status", false);
+            uri.AppendQuery("api-version", _apiVersion, true);
+            request.Uri = uri;
+            request.Headers.Add("Accept", "application/json; odata.metadata=minimal");
+            return message;
+        }
+
+        /// <summary> Returns the current status and synchronization history of a knowledge source. </summary>
+        /// <param name="sourceName"> The name of the knowledge source for which to retrieve status. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="sourceName"/> is null. </exception>
+        public async Task<Response<KnowledgeSourceStatus>> GetStatusAsync(string sourceName, CancellationToken cancellationToken = default)
+        {
```

Comment:
Since this RestClient is internal, we have to add this method in the public [SearchIndexClient.KnowledgeSources.cs](https://github.com/Azure/azure-sdk-for-net/blob/de99c669d49e35bbb79ce3100271c980180c15a3/sdk/search/Azure.Search.Documents/src/Indexes/SearchIndexClient.KnowledgeSources.cs) client.